tests/acct_mgt_test.py

```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/python3
+"""
+ This pytest module checks the results from a running openshift account management
+ microserver using oc (openshift's kubectl equivalent).  This is done instead of mocking
+ openshift, as openshift too frequently changes.
+
+ General commandline format:
+
+ python3 -m pytest amt_test*.py --amurl [acct_mgt_url] --user [username] --passwd [password]
+
+ Note, to do this from apple OSX
+    1) convert the cert and key to a p12 file
+       via: openssl pkcs12 -export -in ./<client_cert> -inkey ./<client_key> -out client.p12
+       openssl pkcs12 -export -in ./acct-mgt-2.crt -inkey ./acct-mgt-2.key -out acct-mgt-2.p12
+
+    2) call curl with
+       curl -v -kv -E ./client.p12:password http://url...
+
+    3) auth_opts can be of the following:
+
+          auth_opts = ["-E","./client_cert/acct-mgt-2.crt", "-key", "./client_cert/acct-mgt-2.key"]
+          auth_opts = ["-cert", r"acct-mgt-2",]
+
+ Initial test to confirm that something is working
+    curl -kv https://acct-mgt.apps.cnv.massopen.cloud/projects/acct-mgt
+    curl -u <user>:<password> -kv https://acct-mgt.apps.cnv.massopen.cloud/projects/acct-mgt
+
+  -- testing with no authentication:
+     python3 -m pytest amt_test*.py --amurl http://acct-mgt..massopen.cloud
+
+  -- testing with basic authentication
+     python3 -m pytest amt_test*.py --amurl https://acct-mgt.massopen.cloud --basic "user:pass"
+"""
+
+import subprocess
+import re
+import time
+import json
+import pprint
+
+
+def get_pod_status(project, pod_name):
+    """This gets the specified pod status from openshift"""
+    result = subprocess.run(
+        ["oc", "-n", project, "-o", "json", "get", "pod", pod_name],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=False,
+    )
+    if result.returncode == 0:
+        result_json = json.loads(result.stdout.decode("utf-8"))
+        print(result_json["status"]["phase"])
+        return result_json["status"]["phase"]
+    print("None")
+    return None
+
+
+#
+def wait_while(project, pod_name, statuses, time_out=300):
+    """
+    This basically waits while a pod is in a set of statuses and possibly times out
+    It is used to wait for openshift to do something with the pod, as in, deleting
+    or starting a pod.
+
+    pass in the following parameter:
+       project:  the namespace of the pod
+       pod_name: the name of the pod
+       statuses: array of statuses
+
+
+    """
+    time_left = time_out
+    time_interval = 5
+    time.sleep(time_interval)
+    status = get_pod_status(project, pod_name)
+    while status in statuses and time_left > 0:
+        time.sleep(time_interval)
```

Comment:
It's better to put your `sleep` at the end of the loop (because this avoids an unnecessary wait if the status is already what you want on your first iteration through the loop).

Reply:
ok, fine, I'll consider getting rid of it since I think it is dead at this point.  I really didn't want this PR to loose focus.

What this automated was me typing 'oc -n <namespace> get pods', while waiting for the pod to go from non-existence to 'container creating' to 'running'.  It saved me a lot of typing while waiting for pods to come up and running.  I have kept it here as I had found it so useful.

The initial wait was to wait for the pod to even enter the 'container creating' stage - it was almost never in the status that I wanted.  So just in case, by some miracle, the pod actually came up and was running after the first wait, I got the status from the pod - and if running is not in the list of statuses it doesn't even enter the loop.  As it didn't make any sense at all to put the wait at the end of loop, after all I had just checked the status, I put it at the beginning of the loop.  And so, yes, I assume that OpenShift is slow (and on our OpenStack cluster, and even on the Power9, 3.11 was slow).  4.x seems to be faster, but it might be that the hardware is better, that is we are running on bare metal with SSDs.  And yes, I did a little loop unrolling.

Reply:
> I did a little loop unrolling

What's the benefit of doing that here?